inference-engine/tests/unit/gna/ngraph/transformations/gna_convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,453 @@
+    }
+
+    return std::make_shared<ngraph::Function>(ngraph::ResultVector{result}, ngraph::ParameterVector{inputParams});
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+void execute_test(const modelType& model, std::shared_ptr<ngraph::Function> function, std::shared_ptr<ngraph::Function> reference_function) {
+    ngraph::pass::Manager manager;
+    manager.register_pass<ngraph::pass::InitNodeInfo>();
+
+    switch (model) {
+    default:
+    case modelType::TranspConvTransp:
+        manager.register_pass<GNAPluginNS::ConvertPadded2ValidConv>();
+        break;
+    case modelType::TranspConvBcastAddTransp:
+        manager.register_pass<GNAPluginNS::ConvertPaddedWithBias2ValidConv>();
+        break;
+    case modelType::TranspConvBcastAddMaxPoolTransp:
+        manager.register_pass<GNAPluginNS::ConvertPaddedWithBiasMaxPool2ValidConv>();
+        break;
+    case modelType::TranspConvBcastAddActTransp:
+        manager.register_pass<GNAPluginNS::ConvertPaddedWithBiasAF2ValidConv>();
+        break;
+    case modelType::TranspConvBcastAddMaxPoolActTransp:
+        manager.register_pass<GNAPluginNS::ConvertPaddedWithBiasMaxPoolAF2ValidConv>();
+        break;
+    case modelType::TranspConvTranspBcastAdd:
+        manager.register_pass<GNAPluginNS::ConvertPaddedTransposedWithBias2ValidConv>();
+        break;
+    case modelType::TranspConvTranspBcastAddAct:
+        manager.register_pass<GNAPluginNS::ConvertPaddedTransposedWithBiasAF2ValidConv>();
+        break;
+    }
+
+    manager.run_passes(function);
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(function, reference_function);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST_P(ConvertPadded2ValidConvTestFixture, CompareFunctions) {
+    execute_test(model, function, reference_function);
+}
+
+INSTANTIATE_TEST_SUITE_P(ConvertPadded2ValidConvTestSuite, ConvertPadded2ValidConvTestFixture,
+    ::testing::Values(
+        std::make_tuple(modelType::TranspConvTransp, ngraph::PartialShape{1, 1, 16, 8}, ngraph::Shape{1, 2}, ngraph::Strides{1, 1},
+            ngraph::CoordinateDiff{0, 2}, ngraph::CoordinateDiff{0, 3}, ngraph::Strides{1, 1},
+            ngraph::Shape{1, 4, 1, 1}, ngraph::Strides{1, 1}, ngraph::Shape{1, 2}, ngraph::op::PadType::EXPLICIT),
+        std::make_tuple(modelType::TranspConvBcastAddTransp, ngraph::PartialShape{1, 1, 16, 8}, ngraph::Shape{1, 2}, ngraph::Strides{1, 1},
+            ngraph::CoordinateDiff{0, 2}, ngraph::CoordinateDiff{0, 3}, ngraph::Strides{1, 1},
+            ngraph::Shape{1, 4, 1, 1}, ngraph::Strides{1, 1}, ngraph::Shape{1, 2}, ngraph::op::PadType::EXPLICIT),
+        std::make_tuple(modelType::TranspConvBcastAddMaxPoolTransp, ngraph::PartialShape{1, 1, 16, 8}, ngraph::Shape{1, 2}, ngraph::Strides{1, 1},
```

Comment:
Consider to shorten.
Many parameters to subsequent `make_tuple` are the same.
eg. `ngraph::Shape{1, 2}, ngraph::Strides{1, 1},
            ngraph::CoordinateDiff{0, 2}, ngraph::CoordinateDiff{0, 3}, ngraph::Strides{1, 1},`

Reply:
That's true, but I left all the parameters to have more flexibility of this test just in case it's needed. I can shorten it, but then again the choice of parameters left would be more or less random and would not make the code cleaner in my opinion. The way it's now all convolution parameters are tweakable in one place.